packages/common/http/src/params.ts

```diff
@@ -78,17 +78,22 @@ function paramParser(rawParams: string, codec: HttpParameterCodec): Map<string,
   }
   return map;
 }
+/**
+ * Encode input string with standard encodeURIComponent and then un-encode specific characters.
+ */
 function standardEncoding(v: string): string {
-  return encodeURIComponent(v)
-      .replace(/%40/gi, '@')
-      .replace(/%3A/gi, ':')
-      .replace(/%24/gi, '$')
-      .replace(/%2C/gi, ',')
-      .replace(/%3B/gi, ';')
-      .replace(/%2B/gi, '+')
-      .replace(/%3D/gi, '=')
-      .replace(/%3F/gi, '?')
-      .replace(/%2F/gi, '/');
+  const tmap = new Map<string, string>([
+    ['40', '@'],
+    ['3A', ':'],
+    ['24', '$'],
+    ['2C', ','],
+    ['3B', ';'],
+    ['2B', '+'],
+    ['3D', '='],
+    ['3F', '?'],
+    ['2F', '/'],
+  ]);
+  return encodeURIComponent(v).replace(/%(\d[a-z0-9])/gi, (_, t) => tmap.get(t) || `%${t}`);
```

Comment:
No need to recreate the string since it is passed as the first parameter.

```suggestion
  return encodeURIComponent(v).replace(/%(\d[a-z0-9])/gi, (s, t) => tmap.get(t) ?? s);
```